src/librustc_mir/transform/const_prop.rs

```diff
@@ -405,13 +405,13 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 }
 
                 let arg = self.eval_operand(arg, source_info)?;
+                let is_release_mode = self.tcx.sess.opts.debugging_opts.mir_opt_level >= 2;
```

Comment:
I'm very sorry about this back and forth. I should have realized it's not the optimization level.

The thing that's actually making the decision is `tcx.sess.overflow_checks()`, so you need to use that.

There's actually 3 levels of optimization decisions going on:

1. `-O`
2. `-C mir-opt-level=2`
3. `-C overflow-checks=false`

The first one impacts the third one if there's no third one specified. The second one doesn't impact anything related to overflow checks.

Also please leave a comment here and at the binop overflow site explaining that we're doing the overflow checks in release mode, too, where there's no assert-terminator doing them for us

Reply:
Okay, added comments and used `overflow_check()`.